### ISSUE TITLE
Problem with Vega in Webpack 5

### ISSUE THREAD

**Issue by blink1073** [jupyterlab/jupyterlab#8655]
## Description
While trying to release 3.0a1, I ran into this issue:

```
    Should not import the named export 'version' (reexported as 'version') from default-exporting module (only default export is available soon)
     @ ./node_modules/vega-embed/build/src/embed.js
     @ ./node_modules/@jupyterlab/vega5-extension/lib/index.js 162:28-48
     @ ./build/index.out.js 108:13-52

    ERROR in ./node_modules/vega/index.js + 608 modules
    Unexpected token (1:71)
    | /* harmony default export */ function __WEBPACK_MODULE_DEFAULT_EXPORT__*(values, valueof) {
    |   if (valueof == null) {
    |     for (let value of values) {
```

I stopped short with the Python release for now, since I only need the JS packages for the extension refactor.

**Comment by blink1073** [comment]
cc @domoritz.   This was with https://github.com/webpack/webpack/releases/tag/v5.0.0-beta.21.  I don't know yet if it is a config issue on our end or not.

**Comment by blink1073** [comment]
Full traceback from the 3.0a2 release:

```
> jlpm run build:prod:release
yarn run v1.21.1
$ ensure-max-old-space webpack --config webpack.prod.release.config.js
(node:97315) [DEP_WEBPACK_COMPILATION_OPTIMIZE_CHUNK_ASSETS] DeprecationWarning: optimizeChunkAssets is deprecated (use Compilation.hook.processAssets instead and use one of Compilation.PROCESS_ASSETS_STAGE_* as stage option)
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:97315) [DEP_WEBPACK_MAIN_TEMPLATE_RENDER_MANIFEST] DeprecationWarning: MainTemplate.hooks.renderManifest is deprecated (use Compilation.hooks.renderManifest instead)
(node:97315) [DEP_WEBPACK_CHUNK_TEMPLATE_RENDER_MANIFEST] DeprecationWarning: ChunkTemplate.hooks.renderManifest is deprecated (use Compilation.hooks.renderManifest instead)
(node:97315) [DEP_WEBPACK_MAIN_TEMPLATE_HASH_FOR_CHUNK] DeprecationWarning: MainTemplate.hooks.hashForChunk is deprecated (use JavascriptModulesPlugin.getCompilationHooks().chunkHash instead)
(node:97315) [DEP_WEBPACK_MAIN_TEMPLATE_GET_ASSET_PATH] DeprecationWarning: MainTemplate.getAssetPath is deprecated (use Compilation.getAssetPath instead)
(node:97315) [DEP_WEBPACK_COMPILATION_NORMAL_MODULE_LOADER_HOOK] DeprecationWarning: Compilation.hooks.normalModuleLoader was moved to NormalModule.getCompilationHooks(compilation).loader
(node:97315) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
(node:97315) [DEP_WEBPACK_MODULE_ID] DeprecationWarning: Module.id: Use new ChunkGraph API
(node:97315) [DEP_WEBPACK_MODULE_UPDATE_HASH] DeprecationWarning: Module.updateHash: Use new ChunkGraph API
(node:97315) [DEP_WEBPACK_CHUNK_MODULES_ITERABLE] DeprecationWarning: Chunk.modulesIterable: Use new ChunkGraph API
(node:97315) [DEP_WEBPACK_CHUNK_GROUP_GET_MODULE_INDEX_2] DeprecationWarning: ChunkGroup.getModuleIndex2 was renamed to getModulePostOrderIndex
Hash: 5d85b76f662e6bda2c3d9b6c51504a951eceb24cebb3c62520480a12feb0
Version: webpack 5.0.0-beta.21
Child
    Hash: 5d85b76f662e6bda2c3d
    Time: 90975 ms
    Built at: 2020-07-07 17:52:27
    87 assets
    Entrypoint main = 9833.7a8fca7f213c7c6855f3.js main.245a46f337df052d2a94.js (260be4f29c0b2ce47480afb23f38f237.ttf 412a43d6840addd683665ec12c30f810.woff2 486853107489520b3265b19b191626f8.svg 4a5d4d855d11fae79cc864bdacb2f479.eot 4c1da237bdae0773309df93b2cd48e09.ttf 5734d789b25228cbafc64a58ae971aca.woff 5c674c9216c06ede2f618aa58ae71116.woff2 66578cdbb6dc01f527a53971051b3e85.svg 6b20949b3a679c30d09f64acd5d3317d.eot 778b1f251bea7412048da95b87bf816f.svg 91a23e8bf2b4b84c39311cb5eb23aaa0.woff2 9a1672a8a8d91fbf82c71f451d495253.eot c65d154888aa166982dac3e72e7380ec.ttf d44ad00c44e46fd29f6126fa7d888cde.woff f3a7d3b5880544a91e9a7e6f8f35d4d2.woff)
    ./node_modules/whatwg-fetch/fetch.js 14.1 KiB [built]
    ./build/index.out.js 28.2 KiB [built]
    ./node_modules/es6-promise/auto.js 179 bytes [built]
    ./node_modules/@jupyterlab/coreutils/lib/index.js 1.05 KiB [built]
    ./build/imports.css 463 bytes [built]
    ./node_modules/@jupyterlab/application/lib/index.js 540 bytes [built]
    ./node_modules/@jupyterlab/javascript-extension/lib/index.js 2.06 KiB [optional] [built]
    ./node_modules/@jupyterlab/json-extension/lib/index.js + 1 modules 7.45 KiB [optional] [built]
    ./node_modules/@jupyterlab/pdf-extension/lib/index.js 6.17 KiB [optional] [built]
    ./node_modules/@jupyterlab/vega5-extension/lib/index.js 5.35 KiB [optional] [built]
    ./node_modules/@jupyterlab/application-extension/lib/index.js + 2 modules 38.8 KiB [built]
    ./node_modules/@jupyterlab/apputils-extension/lib/index.js + 7 modules 48.2 KiB [built]
    ./node_modules/@jupyterlab/celltags-extension/lib/index.js + 3 modules 16.7 KiB [built]
    ./node_modules/@jupyterlab/codemirror-extension/lib/index.js 11.3 KiB [optional] [built]
    ./node_modules/@jupyterlab/completer-extension/lib/index.js + 8 modules 74 KiB [built]
        + 1811 hidden modules

    WARNING in ./node_modules/vega/index.js 20:0-22:24
    Should not import the named export 'version' (reexported as 'version') from default-exporting module (only default export is available soon)
     @ ./node_modules/vega-embed/build/src/embed.js
     @ ./node_modules/@jupyterlab/vega5-extension/lib/index.js 162:28-48
     @ ./build/index.out.js 108:13-52

    ERROR in ./node_modules/vega/index.js + 608 modules
    Unexpected token (1:71)
    | /* harmony default export */ function __WEBPACK_MODULE_DEFAULT_EXPORT__*(values, valueof) {
    |   if (valueof == null) {
    |     for (let value of values) {

Child
    Hash: 9b6c51504a951eceb24c
    Time: 4825 ms
    Built at: 2020-07-07 17:51:01
        Asset      Size
    index.css  15.7 KiB  [emitted]  [name: index]
     index.js   0 bytes  [emitted]  [name: index]
    Entrypoint index = index.css index.js
    ./node_modules/@jupyterlab/theme-dark-extension/style/index.css 39 bytes [built]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/@jupyterlab/theme-dark-extension/style/variables.css 15.3 KiB [built]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/@jupyterlab/theme-dark-extension/style/index.css 472 bytes [built]
        + 1 hidden module
Child
    Hash: ebb3c62520480a12feb0
    Time: 4820 ms
    Built at: 2020-07-07 17:51:01
        Asset      Size
    index.css  14.7 KiB  [emitted]  [name: index]
     index.js   0 bytes  [emitted]  [name: index]
    Entrypoint index = index.css index.js
    ./node_modules/@jupyterlab/theme-light-extension/style/index.css 39 bytes [built]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/@jupyterlab/theme-light-extension/style/variables.css 14.2 KiB [built]
    css ./node_modules/css-loader/dist/cjs.js!./node_modules/@jupyterlab/theme-light-extension/style/index.css 472 bytes [built]
        + 1 hidden module
child_process.js:637
    throw err;
    ^

Error: Command failed: /private/var/folders/y9/qkx7wnj551q65cwhtwb4xrw9s6ybmz/T/jlabrelease_master.WVA2p3rz/jupyterlab/jupyterlab/staging/node_modules/.bin/webpack --config webpack.prod.release.config.js
    at checkExecSyncError (child_process.js:616:11)
    at Object.execFileSync (child_process.js:634:15)
    at Object.<anonymous> (/private/var/folders/y9/qkx7wnj551q65cwhtwb4xrw9s6ybmz/T/jlabrelease_master.WVA2p3rz/jupyterlab/jupyterlab/staging/node_modules/@jupyterlab/buildutils/lib/ensure-max-old-space.js:50:17)
    at Module._compile (internal/modules/cjs/loader.js:1201:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
    at Module.load (internal/modules/cjs/loader.js:1050:32)
    at Function.Module._load (internal/modules/cjs/loader.js:938:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  status: 2,
  signal: null,
  output: [ null, null, null ],
  pid: 97315,
  stdout: null,
  stderr: null
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
child_process.js:655
    throw err;
    ^

Error: Command failed: jlpm run build:prod:release
    at checkExecSyncError (child_process.js:616:11)
    at Object.execSync (child_process.js:652:15)
    at Object.run (/private/var/folders/y9/qkx7wnj551q65cwhtwb4xrw9s6ybmz/T/jlabrelease_master.WVA2p3rz/jupyterlab/buildutils/lib/utils.js:230:32)
    at Object.<anonymous> (/private/var/folders/y9/qkx7wnj551q65cwhtwb4xrw9s6ybmz/T/jlabrelease_master.WVA2p3rz/jupyterlab/buildutils/lib/update-core-mode.js:77:7)
    at Module._compile (internal/modules/cjs/loader.js:1201:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
    at Module.load (internal/modules/cjs/loader.js:1050:32)
    at Function.Module._load (internal/modules/cjs/loader.js:938:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 97181,
  stdout: null,
  stderr: null
}
(node:49996) UnhandledPromiseRejectionWarning: Error: Command failed: node buildutils/lib/update-core-mode.js
    at checkExecSyncError (child_process.js:616:11)
    at Object.execSync (child_process.js:652:15)
    at Object.run (/private/var/folders/y9/qkx7wnj551q65cwhtwb4xrw9s6ybmz/T/jlabrelease_master.WVA2p3rz/jupyterlab/buildutils/lib/utils.js:230:32)
    at Command.<anonymous> (/private/var/folders/y9/qkx7wnj551q65cwhtwb4xrw9s6ybmz/T/jlabrelease_master.WVA2p3rz/jupyterlab/buildutils/lib/publish.js:75:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:49996) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:49996) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
echo
```

**Comment by blink1073** [comment]
It seems like it might be a webpack prod vs development issue.

**Comment by blink1073** [comment]
Yes, minimal repro is here: https://github.com/blink1073/vega-webpack-5

**Comment by blink1073** [comment]
Best we can tell it has to do with the generator operator (`function *`).

**Comment by blink1073** [comment]
We can work around this for now by using a fixed version of webpack in `jupyterlab/staging` and removing the `build:dev:prod:release` check introduced [here](https://github.com/jupyterlab/jupyterlab/pull/8664/files#diff-044502e276dd563a872403f526b9c1eaR84) for now.  We'll want to re-enable that when closing this issue.  I'm going to work on this now.

**Comment by blink1073** [comment]
cf https://github.com/jupyterlab/jupyterlab/pull/8666

**Comment by domoritz** [comment]
Hey, just got to look at this issue. Is there something we should do in Vega or is this purely a bug in webpack?

**Comment by blink1073** [comment]
We're not sure yet, haven't had a chance to dig deeply.

**Comment by blink1073** [comment]
If you could look at  https://github.com/blink1073/vega-webpack-5 and see if you have any ideas for further debugging we would be much obliged :smile:

**Comment by domoritz** [comment]
I won't have the cycles to dig into webpack issues right now but I published Embed 6.9.1, which simplifies how we import the version from JSON, which may resolve the issue.

**Comment by blink1073** [comment]
Here's the verbose output after upgrading embed to 6.9.1 and webpack to beta 22:

```
$ yarn build:prod --display-error-details
yarn run v1.19.1
warning package.json: No license field
$ webpack --mode production --display-error-details
Hash: cf2db53c7ef50c235de4
Version: webpack 5.0.0-beta.22
Time: 7934 ms
Built at: 2020-07-15 010:14:21
2 assets
Entrypoint main = main.js
./index.js 55 bytes [built]
./node_modules/vega-embed/build/src/embed.js 14.4 KiB [built]
./node_modules/fast-json-patch/index.mjs + 3 modules 31.5 KiB [built]
./node_modules/json-stringify-pretty-compact/index.js 2.61 KiB [built]
./node_modules/semver/index.js 1.94 KiB [built]
./node_modules/vega/index.js + 608 modules 1.03 MiB [built]
./node_modules/vega-util/index.js + 47 modules 22.1 KiB [built]
./node_modules/vega-lite/build/src/index.js + 174 modules 724 KiB [built]
./node_modules/tslib/tslib.es6.js 10.4 KiB [built]
./node_modules/vega-schema-url-parser/index.js 414 bytes [built]
./node_modules/vega-themes/build/src/index.js + 9 modules 15.6 KiB [built]
./node_modules/vega-tooltip/build/src/index.js + 5 modules 9.22 KiB [built]
./node_modules/vega-embed/build/src/post.js 847 bytes [built]
./node_modules/vega-embed/build/src/style.js 2.51 KiB [built]
./node_modules/vega-embed/build/src/util.js 717 bytes [built]
    + 309 hidden modules

WARNING in ./node_modules/vega/index.js 20:0-22:24
Should not import the named export 'version' (reexported as 'version') from default-exporting module (only default export is available soon)
    at HarmonyExportImportedSpecifierDependency.getLinkingErrors (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/dependencies/HarmonyImportDependency.js:133:7)
    at HarmonyExportImportedSpecifierDependency._getErrors (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/dependencies/HarmonyExportImportedSpecifierDependency.js:629:15)
    at HarmonyExportImportedSpecifierDependency.getWarnings (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/dependencies/HarmonyExportImportedSpecifierDependency.js:604:15)
    at Compilation.reportDependencyErrorsAndWarnings (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/Compilation.js:1984:24)
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/Compilation.js:1675:10
    at _next2 (eval at create (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:23:1)
    at eval (eval at create (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:50:1)
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/FlagDependencyExportsPlugin.js:289:11
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/neo-async/async.js:2830:7
    at Object.each (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/neo-async/async.js:2850:39)
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/FlagDependencyExportsPlugin.js:268:18
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/neo-async/async.js:2830:7
    at Object.each (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/neo-async/async.js:2850:39)
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/FlagDependencyExportsPlugin.js:46:16
    at Hook.eval [as callAsync] (eval at create (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:46:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/tapable/lib/Hook.js:18:14)
    at Compilation.finish (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/Compilation.js:1668:28)
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/Compiler.js:942:19
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
 @ ./node_modules/vega-embed/build/src/embed.js
 @ ./index.js 1:13-34

ERROR in ./node_modules/vega/index.js + 608 modules
Unexpected token (1:71)
| /* harmony default export */ function __WEBPACK_MODULE_DEFAULT_EXPORT__*(values, valueof) {
|   if (valueof == null) {
|     for (let value of values) {
SyntaxError: Unexpected token (1:71)
| /* harmony default export */ function __WEBPACK_MODULE_DEFAULT_EXPORT__*(values, valueof) {
|   if (valueof == null) {
|     for (let value of values) {
    at NewParser.pp$4.raise (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:2927:15)
    at NewParser.pp.unexpected (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:698:10)
    at NewParser.pp.expect (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:692:28)
    at NewParser.pp$1.parseFunctionParams (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:1303:10)
    at NewParser.pp$1.parseFunction (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:1293:10)
    at NewParser.pp$1.parseFunctionStatement (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:992:17)
    at NewParser.pp$1.parseStatement (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:839:19)
    at NewParser.pp$1.parseTopLevel (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:755:23)
    at NewParser.parse (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:555:17)
    at Function.parse (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/acorn/dist/acorn.js:578:37)
    at Function.parse (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/javascript/JavascriptParser.js:3209:41)
    at ConcatenatedModule._analyseModule (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/optimize/ConcatenatedModule.js:1484:28)
    at ConcatenatedModule.codeGeneration (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/optimize/ConcatenatedModule.js:1016:9)
    at Compilation.codeGeneration (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/Compilation.js:2020:23)
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/Compilation.js:1886:40
    at eval (eval at create (/Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
    at /Users/stslve/workspace/jupyter/vega-webpack-5/node_modules/webpack/lib/optimize/ModuleConcatenationPlugin.js:460:32
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
```

**Comment by blink1073** [comment]
I believe the source in question is:

```
(base) stslve: ~/workspace/jupyter/vega-webpack-5/node_modules (master) 0
$ cat vega-statistics/src/numbers.js
export default function*(values, valueof) {
  if (valueof == null) {
    for (let value of values) {
      if (value != null && value !== '' && (value = +value) >= value) {
        yield value;
      }
    }
  } else {
    let index = -1;
    for (let value of values) {
      value = valueof(value, ++index, values);
      if (value != null && value !== '' && (value = +value) >= value) {
        yield value;
      }
    }
  }
}
```

**Comment by blink1073** [comment]
Yeah, it looks like that file is being pulled in during a production build but not dev.  If I use just that as an entry point `index.js`:

```
$ yarn build
yarn run v1.19.1
warning package.json: No license field
$ webpack --mode development
Hash: a092fb43069b0e252a9f
Version: webpack 5.0.0-beta.22
Time: 56 ms
Built at: 2020-07-15 010:19:54
  Asset      Size
main.js  1.04 KiB  [emitted]  [name: main]
Entrypoint main = main.js
./index.js 503 bytes [built] [failed] [1 error]

ERROR in ./index.js 18:0
Module parse failed: Unexpected token (18:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|     }
|
>

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by blink1073** [comment]
Hmm, it could also be one of the `d3-array/src` files:

```
(base) stslve: ~/workspace/jupyter/vega-webpack-5/node_modules (master) 1
$ search "let value of values"
./d3-array/dist/d3-array.js:51:    for (let value of values) {
./d3-array/dist/d3-array.js:58:    for (let value of values) {
./d3-array/dist/d3-array.js:118:    for (let value of values) {
./d3-array/dist/d3-array.js:127:    for (let value of values) {
./d3-array/dist/d3-array.js:159:    for (let value of values) {
./d3-array/dist/d3-array.js:372:    for (let value of values) {
./d3-array/dist/d3-array.js:393:    for (let value of values) {
./d3-array/dist/d3-array.js:452:    for (let value of values) {
./d3-array/dist/d3-array.js:459:    for (let value of values) {
./d3-array/dist/d3-array.js:513:    for (let value of values) {
./d3-array/dist/d3-array.js:527:    for (let value of values) {
./d3-array/dist/d3-array.js:534:    for (let value of values) {
./d3-array/dist/d3-array.js:570:    for (let value of values) {
./d3-array/dist/d3-array.js:713:    for (let value of values) {
./d3-array/dist/d3-array.js:720:    for (let value of values) {
./d3-array/src/number.js:7:    for (let value of values) {
./d3-array/src/number.js:14:    for (let value of values) {
./d3-array/src/variance.js:7:    for (let value of values) {
./d3-array/src/variance.js:16:    for (let value of values) {
./d3-array/src/maxIndex.js:14:    for (let value of values) {
./d3-array/src/max.js:12:    for (let value of values) {
./d3-array/src/extent.js:17:    for (let value of values) {
./d3-array/src/sum.js:4:    for (let value of values) {
./d3-array/src/sum.js:11:    for (let value of values) {
./d3-array/src/count.js:4:    for (let value of values) {
./d3-array/src/count.js:11:    for (let value of values) {
./d3-array/src/minIndex.js:14:    for (let value of values) {
./d3-array/src/mean.js:5:    for (let value of values) {
./d3-array/src/mean.js:12:    for (let value of values) {
./d3-array/src/min.js:12:    for (let value of values) {
./vega-statistics/src/numbers.js:3:    for (let value of values) {
./vega-statistics/src/numbers.js:10:    for (let value of values) {
```

**Comment by blink1073** [comment]
Perhaps we need to use a babel-loader for JavaScript.

**Comment by blink1073** [comment]
Yep, it was `vega-statistics`, applying this rule allowed the build to run:

```
module.exports = {
    entry: "./index.js",
    module: {
        rules: [{
            test: /vega-statistics\/src\/numbers.js$/,
            use: {
                loader: 'babel-loader',
                options: {
                    presets: ['@babel/preset-env'],
                    plugins: ['@babel/plugin-proposal-object-rest-spread']
                }
            }
        }]
    }
}
```

**Comment by jasongrout** [comment]
I reported this on the webpack 5 feedback issue: https://github.com/webpack/webpack/issues/9802#issuecomment-676826884

**Comment by jasongrout** [comment]
Webpack issue tracking down the bug: https://github.com/webpack/webpack/issues/11353

**Comment by jasongrout** [comment]
Fixed in webpack: https://github.com/webpack/webpack/pull/11354

So hopefully upgrading to the next beta lets us remove the babel framework we introduced.

**Comment by blink1073** [comment]
We'll need to address all the places in the code that reference this issue: https://github.com/jupyterlab/jupyterlab/search?q=8655&unscoped_q=8655

**Comment by blink1073** [comment]
New webpack was released with the fix: https://github.com/webpack/webpack/releases/tag/v5.0.0-beta.29

**Comment by domoritz** [comment]
Btw, I am sending a pull request to Vega to transform all files so we can use any modern JS features but compile to an es6 compatible output: https://github.com/vega/vega/pull/2827.

**Comment by jasongrout** [comment]
Fixed by #8921
